### PR TITLE
refactor(aws): bump aws sdk version

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
At some version, the s3 sdk by default uses the virtual-host style access, which is not supported by MinIO in the default configuration. 

This PR bump aws sdk version to 0.55 and hope to fix the ListObjectV2 error when using s3 compatible object storage.

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
~~- [ ] I have added necessary unit tests and integration tests~~
~~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
~~- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

~~- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.~~

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/object_store/src/object/error.rs [hunk 1]
```diff
@@ -104,20 +106,20 @@ impl ObjectError {
...
-impl From<aws_smithy_http::byte_stream::Error> for ObjectError {
-    fn from(e: aws_smithy_http::byte_stream::Error) -> Self {
-        ObjectErrorInner::S3(e.into()).into()
-    }
-}
+// impl From<aws_smithy_http::byte_stream::Error> for ObjectError {
+//     fn from(e: aws_smithy_http::byte_stream::Error) -> Self {
+//         ObjectErrorInner::S3(e.into()).into()
+//     }
+// }
```
Comment (hzxa21) [inline src/object_store/src/object/error.rs:122]: Is this still needed? If not, we can remove it.

File: Cargo.toml [hunk 2]
```diff
@@ -122,3 +123,5 @@ tokio-stream = { git = "https://github.com/madsim-rs/tokio.git", rev = "ab251ad"
 tokio-retry = { git = "https://github.com/madsim-rs/rust-tokio-retry.git", rev = "95e2fd3" }
 tokio-postgres = { git = "https://github.com/madsim-rs/rust-postgres.git", rev = "4538cd6" }
 madsim-rdkafka = { git = "https://github.com/madsim-rs/madsim.git", rev = "6d342a9" }
+# madsim-aws-sdk-s3 = { path = "../madsim/madsim-aws-sdk-s3" }
```
Comment (hzxa21) [inline Cargo.toml:126]: remove this?

File: src/connector/src/aws_utils.rs [hunk 3]
```diff
@@ -238,9 +223,8 @@ impl AwsConfigV2 {
             .region(region.clone())
             .credentials_provider(cred_provider);
 
-        if let Some(endpoint_uri) = &self.endpoint {
-            let endpoint_val = Endpoint::immutable(endpoint_uri.uri.clone());
-            config_loader = config_loader.endpoint_resolver(endpoint_val);
+        if let Some(endpoint) = &self.endpoint {
+            config_loader = config_loader.endpoint_url(endpoint);
```
Comment (hzxa21) [inline src/connector/src/aws_utils.rs:227]: Have we verified that this change doesn't break the souece?

File: Cargo.toml (hunk 2 above)
Comment (wangrunji0408) [reply Cargo.toml:126]: The adaptation of madsim is in progress. It will be removed when done.

File: src/connector/src/aws_utils.rs (hunk 3 above)
Comment (tabVersion) [reply src/connector/src/aws_utils.rs:227]: lgtm so far, lets trigger main cron ci after merge

**Comment by wangrunji0408** [comment]
The simulation part of aws-sdk-s3 has been updated as well. https://github.com/madsim-rs/madsim/pull/147

**Comment by wcy-fdu** [comment]
@tabVersion  main-cron seems fail, PTAL❤️

**Comment by tabVersion** [comment]
> @tabVersion main-cron seems fail, PTAL❤️

any updates on compactor panic?

**Comment by wcy-fdu** [comment]
> > @tabVersion main-cron seems fail, PTAL❤️
> 
> any updates on compactor panic?

Caused by marge log size, will disable some debug info to avoid.

**Comment by wcy-fdu** [comment]
After changing the timeout threshold up to 32 min, the main-cron **S3 source check on lyvecloud.seagate.com (json parser)** finally passed, and it failed within 25mins timeout threshold.

I think 32 mins is a little more, as @tabVersion has send a PR to turn it up from 20min to 25min, so let's investigate the issue.

**Comment by wcy-fdu** [comment]
> After changing the timeout threshold up to 32 min, the main-cron **S3 source check on lyvecloud.seagate.com (json parser)** finally passed, and it failed within 25mins timeout threshold.
> 
> I think 32 mins is a little more, as @tabVersion has send a PR to turn it up from 20min to 25min, so let's investigate the issue.

Now all check pass, including the main-cron, so I guess previous timeout is due to random network issues. Anyways, this PR is ready for review now, PTAL❤️ @hzxa21

**Review by hzxa21 — APPROVED** [review]
LGTM

**Comment by hzxa21** [comment]
> > After changing the timeout threshold up to 32 min, the main-cron **S3 source check on lyvecloud.seagate.com (json parser)** finally passed, and it failed within 25mins timeout threshold.
> > I think 32 mins is a little more, as @tabVersion has send a PR to turn it up from 20min to 25min, so let's investigate the issue.
> 
> Now all check pass, including the main-cron, so I guess previous timeout is due to random network issues. Anyways, this PR is ready for review now, PTAL❤️ @hzxa21

Did we pass the test without increasing the timeout?

**Review by wangrunji0408 — APPROVED** [review]

**Comment by wcy-fdu** [comment]
> 

Yes, the same as main branch now.

**Pull request merged**